data/tutorials/lg_08_error_handling.md

```diff
@@ -9,62 +9,120 @@ date: 2021-05-27T21:07:30-00:00
 standard library relies heavily upon them.
 
-Exceptions belong to the type `exn` (an extensible sum type):
+The biggest issue with exceptions is that they do not appear in types. One has
+to read the documentation to see that, indeed, `List.find` or `String.sub` are not
+total functions, and that they might fail by raising an exception.
+
+However, exceptions have the great merit of being compiled into efficient
+machine code. When implementing trial and error approaches likely to back-track
+often, exceptions can be used to acheive good performance.
+
+Exceptions belong to the type `exn` which is an [extensible sum type](/releases/latest/manual/extensiblevariants.html).
 
 ```ocaml
+# exception Foo of string;;
 exception Foo of string
 
-let i_will_fail () =
-  raise (Foo "Oh no!")
+# let i_will_fail () =
+  raise (Foo "Oh no!");;
+
+# i_will_fail ();;
+Exception: Foo "Oh no!".
 ```
 
-Here, we add a variant `Foo` to the type `exn`, and create a function
-that will raise this exception. Now, how do we handle exceptions?
-The construct is `try ... with ...`:
+Here, we add a variant `Foo` to the type `exn`, and create a function that will
+raise this exception. Now, how do we handle exceptions? The construct is `try
+... with ...`:
 
 ```ocaml
-let safe_inverse n =
-  try Some (1 / n) with
-    Division_by_zero -> None
-
-let safe_list_find p l =
-  try Some (List.find p l) with
-    Not_found -> None
+# try i_will_fail () with Foo _ -> ();;
+- : unit = ()
 ```
 
-We can try those functions:
+### Predefined Exceptions
+
+The standard library predefines several exceptions, see
+[`Stdlib`](/releases/latest/api/Stdlib.html). Here are a few examples:
 
 ```ocaml
 # 1 / 0;;
 Exception: Division_by_zero.
-# safe_inverse 2;;
-- : int option = Some 0
-# safe_inverse 0;;
-- : int option = None
 # List.find (fun x -> x mod 2 = 0) [1; 3; 5];;
 Exception: Not_found.
-# safe_list_find (fun x -> x mod 2 = 0) [1; 3; 4; 5];;
-- : int option = Some 4
-# safe_list_find (fun x -> x mod 2 = 0) [1; 3; 5];;
-- : int option = None
+# String.sub "Hello world!" 3 (-2);;
+Exception: Invalid_argument "String.sub / Bytes.sub".
+# let rec loop x = x :: loop x
+val loop : 'a -> 'a list = <fun>
+# loop 42;;
+Stack overflow during evaluation (looping recursion?).
+```
+
+Although the last one doesn't look as an exception, it actually is.
+```ocaml
+# try loop 42 with Stack_overflow -> [];;
+- : int list = []
 ```
 
-The biggest issue with exceptions is that they do not appear in types.
-One has to read the documentation to see that, indeed, `Map.S.find`
-or `List.hd` are not total functions, and that they might fail.
+Among them the predefined exceptions of the standard library, the following ones
+are intended to be raised by user written functions:
+```ocaml
+exception Exit
+exception Not_found
+exception Invalid_argument of string
+exception Failure of string
+```
+
+* `Exit` terminates your program with a success status, which is 0 in Unices (they do error values)
```

Comment:
"they do error values" is confusing here. Did you mean to mention that Unix-like systems have processes returning an exit value? Does this even matter in this tutorial?